### ISSUE TITLE
DB::startTransaction assert()-ing

### ISSUE THREAD

**Issue by SchaichAlonso** [silverqx/TinyORM#40]
Currently, TinyORM is [asserting no other transaction](https://github.com/silverqx/TinyORM/blob/e1cd43177c44f5c164b3dae3457409bd257f7511/src/orm/concerns/managestransactions.cpp#L21) on the same connection is in partially commited state when beginTransaction is invoked.

However, the `Q_ASSERT` statements are disabled when [QT_NO_DEBUG](https://doc.qt.io/qt-6/debug.html#debugging-macros) is defined during compilation, i.e. when compiling in release mode. If TinyORM needs it's consumers to explicit distinction between top level transactions and truely nested transactions / savepoints, it should throw an exception when it discoveres savepoints w/o wrapping transactions or transaction nesting rather then relying on `Q_ASSERT` to terminate the application.

**Comment by silverqx** [comment]
This is true, this is one of the first code I wrote and it should be completely refactored to exceptions, i know about this [here](https://github.com/silverqx/TinyORM/blob/e1cd43177c44f5c164b3dae3457409bd257f7511/include/orm/concerns/managestransactions.hpp#L29) .
